### DISCUSSION TITLE
Automated Testing for GraphQL

### DISCUSSION THREAD

**Discussion by daudln** [graphql-python/graphene-django#1500] [General]
### Overview

Automating **GraphQL** testing is essential for maintaining the reliability and stability of our GraphQL APIs. Currently, my process involves manual steps, such as copying queries from Apollo Sandbox into my test functions. This discussion aims to explore ways to streamline and improve my automated testing workflow for GraphQL.

### Current Workflow

My current process for automated testing involves:

- Visiting Apollo Sandbox to copy queries or mutations.
- Pasting the queries into my test functions.
- Running tests to validate API responses.

### Challenges

- Manual effort: Copying queries from Apollo Sandbox is time-consuming.

### Proposed Solutions

- Automation: Explore tools and libraries designed for GraphQL 
- Integration: Integrate GraphQL codegen tool to generate Python code (file) with list of constants of schemas. 

### Discussion Points

- How can I improve the modularity and maintainability of our GraphQL tests?
- What strategies can I implement to reduce manual effort and dependency on external tools?
- Are there any best practices or tools that teams have found effective for automating GraphQL testing?

### Conclusion

By discussing and addressing these challenges, I can enhance our automated testing workflow for GraphQL and ensure the reliability and quality of our APIs.

Let's hear your thoughts and suggestions on how I can optimize our GraphQL testing process!

**Comment by ulgens** [comment]
@daudln This seems like the duplicate of https://github.com/strawberry-graphql/strawberry/discussions/3395 and I think what you are asking doesn't particularly relate to the either libraries. It would be easier for us to answer your question if you can specify how it relates to the tool.
